okhttp/src/main/java/io/grpc/okhttp/OkHttpClientTransport.java

```diff
@@ -476,6 +522,12 @@ public void shutdownNow(Status reason) {
         it.remove();
         entry.getValue().transportReportStatus(reason, false, new Metadata());
       }
+      if (keepAliveManager != null) {
+        // We don't have any active streams. No need to do keepalives any more.
+        // Again, we have to call this inside the lock to avoid the race between onTransportIdle
+        // and onTransportActive.
+        keepAliveManager.onTransportIdle();
```

Comment:
Pending streams are a bit of an edge case. Pending streams used to happen before transportReady() (but that's no longer the case due to changes in TransportSet). Otherwise pendingStreams are only caused by MAX_CONCURRENT_STREAMS. If MAX_CONCURRENT_STREAMS is 0, then it may be a good idea to do pings, since if the connection goes down we'll be permanently hosed as we'll never do writes.